core/sql/parser/sqlparser.y

```diff
@@ -7000,6 +7024,46 @@ rel_subquery_and_as_clause : rel_subquery as_clause
                                CheckModeSpecial1();
                                $$ = $2;
                              }
+with_clause : 
+              TOK_WITH with_clause_elements
+              {
+                 $$ = NULL;
+              }
+              | TOK_WITH TOK_RECURSIVE with_clause_elements
+              {
+                        *SqlParser_Diags << DgSqlCode(-3022) << DgString0("WITH RECURSIVE");
+                        YYERROR;
+              }
+
+with_clause_elements : with_clause_element
+              {
+                 $$ = NULL;
+              }
+              | with_clause_elements ',' with_clause_element
+              {
+                 $$ = NULL;
+              }
+
+with_clause_element : correlation_name TOK_AS '(' query_expression ')'
+                      {
+                         RelRoot *root = new (PARSERHEAP())
+                                            RelRoot($4, REL_ROOT);
+                         $$= new (PARSERHEAP()) RenameTable(root, *$1); 
+
+                         //Duplicated definition of WITH
+                         if(SqlParser_CurrentParser->hasWithDefinition($1) )
+                          {
+                            *SqlParser_Diags << DgSqlCode(-3288)
+                                             << DgString0((*$1).toCharStar());
+                             delete $1;
+                             delete $4;
+                             YYERROR;
+                          }
+
+                          SqlParser_CurrentParser->insertWithDefinition($1 , $$);
+                          delete $1;
+                          delete $4;
```

Comment:
I don't think we want to delete $1 and $4 here, since the RelRoot node points to $4 and the RenameTable node points to $1.